waiter/src/waiter/scheduler.clj

```diff
@@ -447,14 +447,15 @@
   "Takes a map from service -> service instances and performs health checks in parallel. Returns a map of service -> service instances."
   [service->service-instances available? service-id->service-description-fn]
   (let [service->service-instance-futures (start-health-checks service->service-instances available? service-id->service-description-fn)]
-    (loop [[[service {:keys [active-instances] :as instances}] & rest] (seq service->service-instance-futures)
+    (loop [[[service {:keys [active-instances]}] & rest] (seq service->service-instance-futures)
            service->service-instances' {}]
       (if-not service
         service->service-instances'
-        (let [active-instances (doall (map async/<!! active-instances))]
-          (recur rest (assoc service->service-instances'
-                        service
-                        (assoc instances :active-instances active-instances))))))))
+        (recur rest
+               (->> active-instances
+                    (map async/<!!)
+                    doall
```

Comment:
How about `mapv` in place of `map` + `doall`?

Reply:
Done.

Reply:
Removing this change as it breaks the code.